query/src/lib.rs

```diff
@@ -433,6 +467,11 @@ impl Pattern {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Predicate {
+    pub operator: PlainSymbol,
```

Comment:
You're purposely accepting arbitrary predicate operators, right?  You'll constrain that set after parsing?  (Perhaps during interpretation/simplification?)